projects/js-upload-api/src/ClientPkey.ts

```diff
@@ -0,0 +1,209 @@
+
+        if (this.isServerConfigured()) {
+            this._getAuthTokenPromise = this.getAuthToken();
+        }
+    }
+
+    public isServerConfigured(): boolean {
+        console.debug('isServerConfigured', {personalKeyId: this.personalKeyId, _personalKeySecret: this._personalKeySecret, host: this.host});
+        return (this.personalKeyId || '') !== '' && (this._personalKeySecret || '') !== '' && (this.host || '') !== '';
+    }
+
+    public checkStale(personalKeyId: string, personalKeySecret: string, protocol: string, host: string, clientProtocolHostname?: string): boolean {
+        if (this.personalKeyId !== personalKeyId) {
+            console.debug('personalKeyId changed', {currentPersonalKeyId: this.personalKeyId, newPersonalKeyId: personalKeyId}, this);
+            return true;
+        }
+        if (this._personalKeySecret !== personalKeySecret) {
+            console.debug('personalKeySecret changed', {currentPersonalKeySecret: this._personalKeySecret, newPersonalKeySecret: personalKeySecret}, this);
+            return true;
+        }
+        if (this.protocol !== protocol) {
+            console.debug('protocol changed', {currentProtocol: this.protocol, newProtocol: protocol}, this);
+            return true;
+        }
+        if (this.host !== host) {
+            console.debug('host changed', {currentHost: this.host, newHost: host}, this);
+            return true;
+        }
+        if (this.clientProtocolHostname !== clientProtocolHostname) {
+            console.debug('clientProtocolHostname changed', {currentClientProtocolHostname: this.clientProtocolHostname, newClientProtocolHostname: clientProtocolHostname}, this);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Get the authentication token for the current user.
+     * By default, reuses current token if available.
+     * 
+     * @param force If true, forces a new token to be generated; defaults to false
+     * 
+     * @returns The authentication token
+     * 
+     */
+    protected async getAuthToken(force = false): Promise<string> {
+        if (!force && this.authTokenValid()) {
+            return this._token || '';  // workaround ts not recognizing that _token is set
+        }
+
+        //Throw exception if invalid personalKeyId or personalKeySecret
+        if (!this.isServerConfigured()) {
+            console.debug('current config', {personalKeyId: this.personalKeyId, _personalKeySecret: this._personalKeySecret, host: this.host});
+            throw new Error('Invalid personalKeyId or personalKeySecret');
+        }
+
+        if (!force && this._getAuthTokenPromise) {
+            console.debug('reusing outstanding auth promise');
+            return await this._getAuthTokenPromise;
+        }
+
+        console.debug('getAuthToken', {personalKeyId: this.personalKeyId, personalKeySecret: this._personalKeySecret, host: this.host});
+
+        const response = await this.getToApi(
```

Comment:
note that only the personal key and secret are sent to the backend for auth, the org is not included in the request. this is due to the fetch get body issue. am i correct in saying that this means that org sign-ins will not work until the org is re-included in the request?

Reply:
its not clear from reading the nexus code what role any "org" string sent plays in the authentication process in the case of using pkeys. please advise, but will mark this PR done for now.

Reply:
for any specific routes, a good reference is PyGraphistry, which is often the first client of new modes

in this case, for example:
https://github.com/graphistry/pygraphistry/blob/2506b798ec723e906c1c5279f613fe0c37bdbad2/graphistry/arrow_uploader.py#L213